app/models/manageiq/providers/openstack/cloud_manager/host_aggregate.rb

```diff
@@ -1,8 +1,118 @@
 class ManageIQ::Providers::Openstack::CloudManager::HostAggregate < ::HostAggregate
+  include SupportsFeatureMixin
+
+  supports :update_aggregate
+  supports :delete_aggregate
+  supports :add_host
+  supports :remove_host
+
   store :metadata, :accessors => [:availability_zone]
 
   # if availability zone named in metadata exists, return it
   def availability_zone_obj
     AvailabilityZone.find_by_ems_ref_and_ems_id(availability_zone, ems_id)
   end
+
+  def self.create_aggregate(ext_management_system, options)
+    raise ArgumentError, _("ext_management_system cannot be nil") if ext_management_system.nil?
+
+    create_args = {:name => options[:name]}
+    if options[:availability_zone]
+      create_args[:availability_zone] = options[:availability_zone]
+    end
+    aggregate = nil
+    metadata = {}
+
+    connection_options = {:service => "Compute"}
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      aggregate = service.aggregates.create(create_args)
+      if aggregate.availability_zone
+        metadata[:availability_zone] = aggregate.availability_zone
+      end
+    end
+    create(:name                  => options[:name],
```

Comment:
I'd use `create!` here, otherwise `create` will just return a AR object, which happens to be `valid? => false`
`create!` will raise an exception